lib/package.gi

```diff
@@ -2202,10 +2226,33 @@ InstallGlobalFunction( ValidatePackageInfo, function( info )
     TestMandat( record, "Version",
         x -> IsString(x) and 0 < Length(x) and x[1] <> '=',
         "a nonempty string that does not start with `='" );
-    TestMandat( record, "Date",
-        x -> IsString(x) and Length(x) = 10 and x{ [3,6] } = "//"
-                 and ForAll( x{ [1,2,4,5,7,8,9,10] }, IsDigitChar ),
-        "a string of the form `dd/mm/yyyy'" );
+
+    # check if the date is valid
+    CheckDateValidity := function(x)
+      local date;
+
+      if not ( IsString(x) and Length(x) = 10
+        and ForAll( x{ [1,2,4,5,7,8,9,10] }, IsDigitChar )
+        and ( x{ [3,6] } = "//" or x{ [5,8] } = "--" ) ) then
+        return false;
+      elif x{ [3,6] } = "//" then # for the old format split at '/'
+        date := List( SplitString( x, "/" ), Int);
+      elif x{ [5,8] } = "--" then # for the yyyy-mm-dd format split at '-'
+        date := List( SplitString( x, "-" ), Int);
+        date := Permuted( date, (1,3) ); # sort such that date=[dd,mm,yyyy]
+      fi;
+      return date[2] in [1..12] and date[3] >= 1999 # GAP 4 appeared in 1999
+          and date[1] in [1..DaysInMonth( date[2], date[3] )];
+    end;
+
+    TestMandat( record, "Date", CheckDateValidity, Concatenation( "a string of the form yyyy-mm-dd or dd/mm/yyyy",
+    " that represents a date since 1999") );
+
+    # If the date is in the format `dd/mm/yyyy` a message is printed
+    if IsBound( record.Date ) and  CheckDateValidity( record.Date ) and record.Date{ [3,6] } = "//" then
+       Info( InfoPackageLoading, 2, Concatenation( record.PackageName, ": Please be advised to change the date format to `yyyy-mm-dd`") );
```

Comment:
Do we really want to print this right now? Perhaps better to comment that code out now, at least until we are sure all relevant tools have been updated. Things to check include:
- [ ] the package distribution
- [x] AutoDoc (already updated by me for ISO dates)
- [ ] GitHubPagesForGAP -- the `update.g` script parses the date (I *think* I already prepared that for ISO dates some time ago, but (a) this is untested and (b) some packages might still be using an older version of the `update.g` script

Reply:
I've made a PR for the package distro although I don't know if that is all which has to be changed. 
I can comment it out, but I already have it only for InfoLevel 2 so that it doesn't get printed all the time or am I misunderstanding the concept here?

Reply:
Are you referring to https://github.com/gap-system/gap-distribution/pull/78 ? I commented on that last week :-)

Reply:
Ah sorry I wasn't notified by GitHub that something happened there. I'll take care of it thanks! (also GitHub didn't tell me that you made the comment above telling me about the comment in the other repo. Sometimes I'm a bit confused about what it will notify me of and what not)